hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -217,9 +218,19 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       // creation and key commit, old versions will be just overwritten and
       // not kept. Bucket versioning will be effective from the first key
       // creation after the knob turned on.
+      boolean isPreviousCommitHsync = false;
       Map<String, RepeatedOmKeyInfo> oldKeyVersionsToDeleteMap = null;
       OmKeyInfo keyToDelete =
           omMetadataManager.getKeyTable(getBucketLayout()).get(dbOzoneKey);
+      if (null != keyToDelete) {
+        final String clientIdString
+            = String.valueOf(commitKeyRequest.getClientID());
+        isPreviousCommitHsync = java.util.Optional.ofNullable(keyToDelete)
```

Comment:
Thanks @sumitagrawl .

I see you are using `keyToDelete` (from `KeyTable`) instead of `omKeyInfo` (from `OpenKeyTable`) here from the [previous](https://github.com/apache/ozone/pull/5167#discussion_r1290061020) patch file. Looks like this should work.

This correctly forces subsequent hsync OR final commit to fall into the first code path (line 263). And I assume any allocated but committed blocks will still be cleaned up by `uncommitted` logic below?

Reply:
Yes, it will hit the logic and will get Hsync flag from keyToDelete as verified. This is handled with another flag isPreviousCommitHsync. And Uncommited will be cleaned as normal HSync flag is false.